### DISCUSSION TITLE
What is a slot?

### DISCUSSION THREAD

**Discussion by markusschloesser** [markusschloesser/MackieC4_P3#24] [Q&A] [answered]
what is a "slot"?
I know scenes, clips, tracks, devices etc but NOT a slot. Refers to https://github.com/markusschloesser/MackieC4_P3/blob/489c46748fbe2fb6417514d22645a0ae64f706cb/MackieC4.py#L451 and
https://github.com/markusschloesser/MackieC4_P3/blob/489c46748fbe2fb6417514d22645a0ae64f706cb/MackieC4.py#L661
and what is the `tmptrack` for?
Asking cos it throws errors sometimes

**Comment by BilldarBagdar** [comment, marked as answer]
My **guess** is that Tracks have "clip slots" where clips "live" (or not). (and that "slot" is just shorthand for "clip slot")  In the Session view, every "scene" is a row of "clip slots" across all tracks in the session.  

`tmptrack` in `tmptrack = LiveUtils.getTrack(tid)`  is just another "local variable name".  If this assignment is "throwing errors" it is likely because the `tid` value (the "track ID" value?) does not point to a valid "track address" in the LOM.  (If there are 3 tracks in the session `LiveUtils.getTrack(3)` would "error out" because the 3 tracks are at indexes 0, 1, and 2 but `getTrack(3)` asks for the track "one past the end" and is an error.)

**Comment by markusschloesser** [comment]
answered (and clips and slots are out-commented anyway currently)
